rust/src/detect/byte_extract.rs

```diff
@@ -205,7 +205,7 @@ fn parse_byteextract(input: &str) -> IResult<&str, SCDetectByteExtractData, Rule
                 let mult = val
                     .parse::<u32>()
                     .map_err(|_| make_error(format!("invalid multiplier value: {}", val)))?;
-                if mult == 0 || mult > u16::MAX.into() {
+                if mult == 0 || mult > u16::MAX as u32 {
```

Comment:
These and the like should probably have never used this .into() syntax anyways, as we know the type on the other side.

Reply:
> These and the like should probably have never used this .into() syntax anyways, as we know the type on the other side.

This uses `into()` because we (and the rust compiler) know the type on the other side and know how to convert it safely (without losing precision)

`as` is not safe in all cases (here it is safe)

Reply:
`u32::from(u16::MAX)` ?

Reply:
Indeed, because `deranged` overrode `==` with an integer-like type, right ?